lib/api/core/models/repositories/profileRepository.js

```diff
@@ -0,0 +1,98 @@
+module.exports = function (kuzzle) {
+  var
+    q = require('q'),
+    Profile = require('../security/profile'),
+    Repository = require('./repository');
+
+
+  function ProfileRepository () {
+    this.profiles = {};
+  }
+
+  ProfileRepository.prototype = new Repository(kuzzle, {
+    collection: '_kuzzle/profiles',
+    ObjectConstructor: Profile,
+    cacheEngine: kuzzle.services.list.userCache
+  });
+
+
+  ProfileRepository.prototype.loadProfile = function (profileKey) {
+    var
+      deferred = q.defer();
+
+    if (this.profiles[profileKey]) {
+      deferred.resolve(this.profiles[profileKey]);
+    }
+    else {
+      this.loadOneFromDatabase(profileKey)
+        .then(function (result) {
+          var
+            data,
+            p = new Profile();
+
+          if (result) {
+            // we got a profile from the database, we can use it
+            deferred.resolve(result);
+          }
+          else if (kuzzle.config.defaultUserProfiles[profileKey]) {
+            // no profile found in db but we have a default config for it
+            data = kuzzle.config.defaultUserProfiles[profileKey];
+            deferred.resolve(this.hydrate(p, data));
```

Comment:
Since hydrate() returns a promise, shouldn't this be something along these lines?

```javascript
this.hydrate(p, data)
  .then(instance => deferred.resolve(instance)
  .catch(error => deferred.reject(error));
```